webpack/assets/javascripts/react_app/common/I18n.js

```diff
@@ -0,0 +1,42 @@
+import Jed from 'jed';
+import { deprecateObjectProperty } from '../../foreman_tools';
+
+class I18n {
+  constructor(locales) {
+    this.cheveronPrefix = window.I18N_MARK ? '\u00BB' : '';
+    this.cheveronSuffix = window.I18N_MARK ? '\u00AB' : '';
+    this.translate = this.translate.bind(this);
+    this.ngettext = this.ngettext.bind(this);
+    this.initLocale();
+    this.jed = new Jed(locales[this.locale]);
+  }
+  initLocale() {
+    const locale = document.getElementsByTagName('html')[0].lang;
+    this.locale = locale.length === 0 ? 'en' : locale.replace(/-/g, '_');
+  }
+  translate(...args) {
+    return `${this.cheveronPrefix}${this.jed.gettext(...args)}${this.cheveronSuffix}`;
+  }
+  ngettext(...args) {
+    return `${this.cheveronPrefix}${this.jed.ngettext(...args)}${this.cheveronSuffix}`;
+  }
+}
+
+const emptyLocales = {
+  en: { domain: 'app', locale_data: { app: { '': {} } } },
+};
+
+const locales = window.locales || emptyLocales;
+const i18n = new I18n(locales);
+export const { translate, ngettext, jed } = i18n;
+export const { sprintf } = jed;
+
+window.__ = translate;
+window.n__ = ngettext;
+window.Jed = i18n.jed;
+window.i18n = i18n.jed;
+
+deprecateObjectProperty(window, 'i18n', 'tfm.i18n');
+deprecateObjectProperty(window, 'Jed', 'tfm.i18n');
```

Comment:
1. it should be
```js
deprecateObjectProperty(window, 'Jed', 'tfm.i18n.jed');
```
2. note that you didn't add any deprecations to:
```window.__``` and ```window.n__```

Reply:
That's in purpose, in legacy js code I guess it more naturally to use `__()` then `tfm.i18.translate()` so I kept that behavior undeprecated